### ISSUE TITLE
Question: WatchConfig

### ISSUE THREAD

**Issue by tkliner** [spf13/viper#534]
In my configuration I have set WatchConfig() and in main file, when I do not specify (for example, in func init ()) this code:

```
config.OnConfigChange(func(e fsnotify.Event) {
...
}
```
 the program will crash when changing the configuration file:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x14163c6]

goroutine 51 [running]:
github.com/spf13/viper.(*Viper).WatchConfig.func1.1(0xc4201d2000, 0xc4200c43c0, 0x36, 0xc4201a60f0)
        ~/go/src/github.com/spf13/viper/viper.go:292 +0x2d6
created by github.com/spf13/viper.(*Viper).WatchConfig.func1
        ~/go/src/github.com/spf13/viper/viper.go:281 +0x1f6
exit status 2
The terminal process terminated with exit code: 1
```
If i enter the code in init function:

```
config.OnConfigChange(func(e fsnotify.Event) {
...
}
```

the program will not crash. Is this the right behavior? Respectively, if I just want to change the key of a key in the configuration, why do I need to provide this code?

**Comment by movergan** [comment]
Just encountered the same issue.

I run WatchConfig() as a part of config load function and as soon as config changes program crash, 

`func LoadConfig(confname string, Config interface{}) {
	var runtimeViper = viper.New()
	runtimeViper.SetConfigType("yaml")
	runtimeViper.SetConfigName(confname)   // name of config file (without extension)
	runtimeViper.AddConfigPath("./config") // path to look for the config file in
	runtimeViper.AddConfigPath(".")        // optionally look for config in the working directory
	err := runtimeViper.ReadInConfig()     // Find and read the config file
	if err != nil {                        // Handle errors reading the config file
		panic(fmt.Errorf("Fatal error config file: %s \n", err))
	}
	runtimeViper.WatchConfig()
	runtimeViper.Unmarshal(Config)
}`

**Comment by lnashier** [comment]
I have faced same issue in past. Callback has to be registered since Viper is not checking/validating for presence of callback.

**Comment by andrewstuart** [comment]
This should be fixed by #568. :slightly_smiling_face:
